docs/src/pages/docs/crocks/Tuple.md

```diff
@@ -0,0 +1,480 @@
+that can be represented.
+
+The resulting `n-Tuple` constructor is parameterized as an `n-Functor`, where
+each parameter can vary from instance to instance.
+
+```javascript
+import Tuple from 'crocks/Tuple'
+
+// Triple :: 3-Tuple
+const Triple =
+  Tuple(3)
+
+// Quad :: 4-Tuple
+const Quad =
+  Tuple(4)
+
+Triple(false, true, 45)
+//=> 3-Tuple (false, true, 45)
+
+Quad({ a: true }, [ 1, 2, 3 ], 60, 'string')
+//=> 4-Tuple ({ a: true }, [ 1, 2, 3 ], 60, "string")
+```
+
+</article>
+
+<article id="topic-instance">
+
+## Instance Methods
+
+#### equals
+
+```haskell
+n-Tuple *...n ~> a -> Boolean
+```
+
+Used to compare the values of two `n-Tuple` instances by value.`equals` takes
+any given argument and will return a `true` if passed an `n-Tuple` of the same
+size with match values in the same positions as the `n-Tuple` `equals` was
+run on. If the provided argument is not an `n-Tuple` of the same type or the
+underlying values are not equal, then `equals` will return `false`.
+
+
+```javascript
+import Tuple from 'crocks/Tuple'
+
+// Pair :: 2-Tuple
+const Pair =
+  Tuple(2)
+
+// Triple :: 3-Tuple
+const Triple =
+  Tuple(3)
+
+Pair(1, false)
+  .equals(Pair(1, false))
+//=> true
+
+Triple(1, false, [ 1, 2 ])
+  .equals(Triple(1, false, [ 1, 2 ]))
+//=> true
+
+Triple(1, false, [ 1, 2 ])
+  .equals(Triple(1, true, [ 3, 4 ]))
+//=> false
+
+Pair(1, false)
+  .equals(Triple(1, false, [ 1, 2 ]))
+//=> false
+```
+
+#### concat
+
+```haskell
+Semigroup s1 => 1-Tuple s1 ~> 1-Tuple s1 -> 1-Tuple s1
+Semigroup s1, s2 => 2-Tuple s1 s2 ~> 2-Tuple s1 s2 -> 2-Tuple s1 s2
+...
+```
+
+When a given `n-Tuple` contains `Semigroup` instances in all defined `n`
+positions. It can be concatenated with another `n-Tuple` instance containing
+instances of the same `Semigroup`s in the same positions as the `n-Tuple`
+instance it is being concatenated with.
+
+`concat` will be called on each contained `Semigroup` instance with the instance
+corresponding to the other `n-Tuple` instance. The result of each concatenation
+will be provided in a new `n-Tuple` instance of the same `Semigroup`s.
+
+```javascript
+import Tuple from 'crocks/Tuple'
+import Sum from 'crocks/Sum'
+
+// Triple :: 3-Tuple
+const Triple =
+  Tuple(3)
+
+// Unary :: 1-Tuple
+const Unary =
+  Tuple(1)
+
+Triple([ 1, 3 ], Sum(10), Sum(1))
+  .concat(Triple([ 4 ], Sum.empty(), Sum(9)))
+//=> 3-Tuple( [ 1, 3, 4 ], Sum 10, Sum, 10 )
+
+Unary([ 10 ])
+  .concat(Unary([ 10 ]))
+//=> 1-Tuple( [ 10, 10 ] )
+```
+
+#### map
+
+```haskell
+1-Tuple a ~> (a -> b) -> 1-Tuple b
+2-Tuple a b ~> (b -> c) -> 2-Tuple a c
+...
+```
+
+Used to lift a single function into a given `n-Tuple` to map the rightmost
+value. `map` takes a function `(a -> b)` and will return a
+new `n-Tuple` instance with the result of mapping the rightmost value from its
+original `a` type to the resulting `b`.
+
+This method will only apply to the rightmost value, if the need to map one or
```

Comment:
Here we can use the `.` instead of `,`. And also I think we can restructure this a bit something like:
> This method will only apply to the rightmost value. 
> [`mapAll`](#mapall) can be used to map one or many
> of the values in the various other positions.